rpcs3/Emu/NP/np_notifications.cpp

```diff
@@ -266,4 +267,135 @@ namespace np
 		auto& sigh = g_fxo->get<named_thread<signaling_handler>>();
 		sigh.send_information_packets(addr_p2p, port_p2p, npid_p2p);
 	}
+
+	void np_handler::generic_gui_notification_handler(std::vector<u8>& data, std::string_view name, s32 notification_type)
+	{
+		vec_stream noti(data);
+		auto* update_info = noti.get_flatbuffer<MatchingRoomStatus>();
+
+		if (noti.is_error())
+		{
+			rpcn_log.error("Received faulty %s notification", name);
+			return;
+		}
+
+		std::lock_guard lock(gui_notifications.mutex);
+
+		if (!gui_notifications.current_gui_ctx_id)
+			return;
+
+		// Should the callback signaled depend on the room id and should we track if the context is the one that created/joined/left/etc the room?
+		auto ctx = get_matching_context(gui_notifications.current_gui_ctx_id);
+		ensure(ctx);
+		u32 req_id = gui_notifications.counter_req_id++;
+
+		event_data edata(np_memory.allocate(MAX_SceNpMatchingRoomStatus_SIZE), sizeof(SceNpMatchingRoomStatus), MAX_SceNpMatchingRoomStatus_SIZE);
+		auto* room_status = reinterpret_cast<SceNpMatchingRoomStatus*>(edata.data());
+		MatchingRoomStatus_to_SceNpMatchingRoomStatus(edata, update_info, room_status);
+		np_memory.shrink_allocation(edata.addr(), edata.size());
+
+		extra_nps::print_SceNpMatchingRoomStatus(room_status);
+
+		switch (notification_type)
+		{
+		case SCE_NP_MATCHING_EVENT_ROOM_UPDATE_NEW_MEMBER:
+			gui_cache.add_member(room_status->id, room_status->members.get_ptr(), true);
+			break;
+		case SCE_NP_MATCHING_EVENT_ROOM_UPDATE_MEMBER_LEAVE:
+			gui_cache.del_member(room_status->id, room_status->members.get_ptr());
+			break;
+		case SCE_NP_MATCHING_EVENT_ROOM_DISAPPEARED:
+			gui_cache.del_room(room_status->id);
+			break;
+		case SCE_NP_MATCHING_EVENT_ROOM_UPDATE_OWNER_CHANGE:
+			gui_cache.add_member(room_status->id, room_status->members.get_ptr(), false);
+			gui_cache.add_member(room_status->id, room_status->members->next.get_ptr(), false);
+			break;
+		case SCE_NP_MATCHING_EVENT_ROOM_KICKED:
+			gui_cache.del_room(room_status->id);
+			break;
+		default:
+			fmt::throw_exception("Unexpected notification type in generic_gui_notification_handler: 0x%08X", notification_type);
+			break;
+		}
+
+		gui_notifications.list.emplace(std::make_pair(gui_notifications.current_gui_ctx_id, req_id), gui_notification{.event = notification_type, .edata = std::move(edata)});
+		ctx->queue_callback(req_id, notification_type, 0);
```

Comment:
does this not need a wakey like the one below?

Reply:
No, only quickmatch has a timeout which spawns a thread.